test/calc.js

```diff
@@ -196,4 +197,24 @@ describe('shouldUpdate', () => {
     const result = shouldUpdate(100, 110, propDefaults, state)
     expect(result.action).to.equal('unpin')
   })
+
+  it("should return an action of 'pin' if props.pin is set and not pinned, yet", () => {
+    const state = {
+      height: 100,
+      state: 'something',
+    }
+
+    const result = shouldUpdate(100, 110, { ...propDefaults, pin: true }, state)
+    expect(result.action).to.equal('pin')
+  })
+
+  it("should return an action of 'none' if props.pin is set and already pinned", () => {
+    const state = {
+      height: 100,
+      state: 'pined',
+    }
+
+    const result = shouldUpdate(100, 110, { ...propDefaults, pin: true }, state)
+    expect(result.action).to.equal('pin')
```

Comment:
This is wrong as well, the test should expect `none` as the title says. However, in combination with the typo from above this test succeeds by accident.